app/code/community/Bolt/Boltpay/sql/bolt_boltpay_setup/upgrade-1.4.0-2.0.0.php

```diff
@@ -33,11 +33,14 @@
 $statusTable = $installer->getTable('sales/order_status');
 $statusStateTable = $installer->getTable('sales/order_status_state');
 
-$pendingBoltStatusDoesntExists = !is_object($connection->query("SELECT status FROM $statusTable WHERE status = 'pending_bolt'")->fetchObject());
-$pendingBoltMappingDoesntExists = !is_object($connection->query("SELECT status FROM $statusStateTable WHERE status = 'pending_bolt' AND state = 'pending_payment'")->fetchObject());
+$pendingBolt = Bolt_Boltpay_Model_Payment::TRANSACTION_PRE_AUTH_PENDING;
```

Comment:
In general I'm not big fan of using externally-defined value in migration. If we for some reason change the value of this constant, it'll change content of migration file, causing very hard to detect bug.

Reply:
@daisy1754 
The PR has changed and I see that you did not review the existing code.  Please review the existing code.

Reply:
I'm confused. what do you mean by existing code? as far as I can tell only change made in this PR after latest commit is using constants `Bolt_Boltpay_Model_Payment::TRANSACTION_PRE_AUTH_PENDING` and `Bolt_Boltpay_Model_Payment:: TRANSACTION_PRE_AUTH_CANCELED` in migration, and that's exactly what I mention in my previous comment.

Reply:
@daisy1754 Actually, the major change was `removed order cleanup code and will trust in hooks` https://github.com/BoltApp/bolt-magento1/pull/385/commits/5e876f22a2c541b1793ade50bc766a673ca13e38

> In general I'm not big fan of using externally-defined value in migration. If we for some reason change the value of this constant, it'll change content of migration file, causing very hard to detect bug.

I did not see this above comment.  :O

Hmm.  Actually, I thought over this quite a bit prior to making the change thinking just as you had written above.  I believe that for poorly written migration code, you are correct.    However, for properly written migration code for updating the value which should always follow the pattern

- *if old row values exist* DELETE
- *if new row values do not exist* INSERT

Or if supported, UPSERT or in MySQL INSERT IGNORE, maybe this shouldn't be a problem.  I was thinking that an author updating table constants should be extremely conscious of this.

Ok.  Sounds good to me.  Let's use hard coded values.